Don't require user to create profile manually even before guix-daemon runs.

Default install should provide working basic environment.

Revert "build: Don't embed absolute paths in .service and .conf service files."
This reverts commit 613d0895b92c677e0639d5e77c55043e38e020c8.
diff --git a/etc/guix-daemon.conf.in b/etc/guix-daemon.conf.in
index 09c70cde3..8f833cb69 100644
--- a/etc/guix-daemon.conf.in
+++ b/etc/guix-daemon.conf.in
@@ -9,4 +9,4 @@ stop on runlevel [016]
 
 task
 
-exec @localstatedir@/guix/profiles/per-user/root/guix-profile/bin/guix-daemon --build-users-group=guixbuild
+exec @bindir@/guix-daemon --build-users-group=guixbuild
diff --git a/etc/guix-daemon.service.in b/etc/guix-daemon.service.in
index 988cf90c0..7a3f02b27 100644
--- a/etc/guix-daemon.service.in
+++ b/etc/guix-daemon.service.in
@@ -6,7 +6,7 @@
 Description=Build daemon for GNU Guix
 
 [Service]
-ExecStart=@localstatedir@/guix/profiles/per-user/root/guix-profile/bin/guix-daemon --build-users-group=guixbuild
+ExecStart=@bindir@/guix-daemon --build-users-group=guixbuild
 Environment=GUIX_LOCPATH=/root/.guix-profile/lib/locale
 RemainAfterExit=yes
 StandardOutput=syslog
diff --git a/etc/guix-publish.conf.in b/etc/guix-publish.conf.in
index 241c59455..498fa295b 100644
--- a/etc/guix-publish.conf.in
+++ b/etc/guix-publish.conf.in
@@ -9,4 +9,4 @@ stop on runlevel [016]
 
 task
 
-exec @localstatedir@/guix/profiles/per-user/root/guix-profile/bin/guix publish --user=nobody --port=8181
+exec @bindir@/guix publish --user=nobody --port=8181
diff --git a/etc/guix-publish.service.in b/etc/guix-publish.service.in
index 8aaf09e3c..fc4e3c21f 100644
--- a/etc/guix-publish.service.in
+++ b/etc/guix-publish.service.in
@@ -6,7 +6,7 @@
 Description=Publish the GNU Guix store
 
 [Service]
-ExecStart=@localstatedir@/guix/profiles/per-user/root/guix-profile/bin/guix publish --user=nobody --port=8181
+ExecStart=@bindir@/guix publish --user=nobody --port=8181
 Environment=GUIX_LOCPATH=/root/.guix-profile/lib/locale
 RemainAfterExit=yes
 StandardOutput=syslog
diff --git a/nix/local.mk b/nix/local.mk
index 9e0c457be..eb70d266f 100644
--- a/nix/local.mk
+++ b/nix/local.mk
@@ -190,7 +190,7 @@ nodist_systemdservice_DATA = etc/guix-daemon.service etc/guix-publish.service
 etc/guix-%.service: etc/guix-%.service.in	\
 			 $(top_builddir)/config.status
 	$(AM_V_GEN)$(MKDIR_P) "`dirname $@`";	\
-	$(SED) -e 's|@''localstatedir''@|$(localstatedir)|' <	\
+	$(SED) -e 's|@''bindir''@|$(bindir)|' <	\
 	       "$<" > "$@.tmp";		\
 	mv "$@.tmp" "$@"
 
@@ -201,7 +201,7 @@ nodist_upstartjob_DATA = etc/guix-daemon.conf etc/guix-publish.conf
 etc/guix-%.conf: etc/guix-%.conf.in	\
 			 $(top_builddir)/config.status
 	$(AM_V_GEN)$(MKDIR_P) "`dirname $@`";	\
-	$(SED) -e 's|@''localstatedir''@|$(localstatedir)|' <	\
+	$(SED) -e 's|@''bindir''@|$(bindir)|' <	\
 	       "$<" > "$@.tmp";		\
 	mv "$@.tmp" "$@"
 
